### PR TITLE
feat(ui): enabling feature flags for various UI features

### LINKED ISSUE #21164: Enable UI Features for OSS
Now that we are using UI assets from the `ui` repository (implemented in #21090), we have access to the latest UI features for OSS.

Many of these UI features can't be supported by the OSS backend just yet, but quite a few of them can. We'd like to turn these supported UI features on by enabling a feature flag on the OSS backend to let the frontend know to display the feature - this can be done by updating the OSS `flags.yml`. Ideally, the feature flags would be removed from the frontend code, but the flag removal process currently lags behind the point at which the UI features are "ready" to be included in an OSS release. We need to turn on the feature flags so the latest OSS releases have the latest supported UI features.

Currently, the following features are fully supported by the OSS backend, and can be turned on via feature flag:

| Description                        | Flag                      | OSS Work Needed                                                                                                                                    |
|------------------------------------|---------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
| Axis Ticks Generator               | axisTicksGenerator        | Done in #19850                                                                                          |
| Band Plot Type                     | bandPlotType              | Done in [fecc4fd](https://github.com/influxdata/influxdb/commit/fecc4fd533d948a12966c335da62b5aee4cc8e47)                                                     |
| UI CSV Uploader                    | csvUploader               | None - UI only, parses a CSV client-side and uses existing OSS API                                                                                 |
| Editable Telegraf Configurations   | editTelegrafs             | None - UI only, uses existing OSS API                                                                                                              |
| Legend Orientation in the UI       | legendOrientation         | Done in #19584                                                                                          |
| Mosaic Graph Type                  | mosaicGraphType           | Done in #19195 & [b6275bf](https://github.com/influxdata/influxdb/commit/b6275bff9e6af48e43c795b3b14fe9ba0b8a4f0b) |
| Default Monaco Selection to EOF    | cursorAtEOF               | None - UI only                                                                                                                                     |
| Refresh Single Cell                | refreshSingleCell         | None - UI only, uses existing OSS API                                                                                                              |
| Type Ahead Dropdowns for Variables | typeAheadVariableDropdown | None - UI only                                                                                                                                     |

### PR DESCRIPTION
**Pull request by wbaker85**
Partially closes #21164

This PR turns "on" the feature flags for various UI features, including:
- Axis Ticks Generator (for controlling the number of axis ticks on plots)
- Band Plot Type
- Mosaic Plot Type
- UI CSV Uploader (upload annotated CSVs to a bucket)
- Editable Telegraf Configurations (edit telegraf configs from the UI)
- Legend Orientation in the UI (customize the appearance of the legend)
- Refresh Single Cell (instead of refreshing the entire dashboard's cells)

The two that aren't being turned on with this PR are listed below. To turn these on, we will need to move on from using our "legacy" e2e tests in `influxdb/ui/cypress` and be able to use the latest and greatest e2e tests from the `ui` repo, the implementation of which is described in [influxdata/ui#1007](https://github.com/influxdata/ui/issues/1007).
- Default Monaco Selection to EOF (puts the cursor at the end of the line when editing scripts)
- Type Ahead Dropdowns for Variables

### CODE REVIEW

File: flags.yml [hunk 1]
```diff
@@ -119,3 +119,59 @@
...
+
+- name: Legend Orientation in the UI
+  description: Change the appearance of the legend
+  key: legendOrientation
+  default: true
+  contact: Monitoring Team
+  expose: true
+  lifetime: temporary
+
+- name: Default Monaco Selection to EOF
```
Comment (danxmoran) [inline flags.yml:155]: Is this one related to the weird behavior we see on Windows?

Comment (wbaker85) [reply flags.yml:155]: Nope, that's actually fixed in [`ui/#1129`](https://github.com/influxdata/ui/pull/1129).

File: ui/cypress/e2e/explorer.test.ts [hunk 2]
```diff
@@ -735,7 +735,7 @@ describe('DataExplorer', () => {
         cy.getByTestID('time-machine-submit-button').click()
         cy.getByTestID('cog-cell--button').click()
         cy.getByTestID('select-group--option')
-          .contains('Custom')
+          .last()
```
Comment (wbaker85) [inline ui/cypress/e2e/explorer.test.ts:738]: Had to make this very small tweak to our existing OSS e2e tests. With the ability to configure the tick marks enabled by the flag `axisTicksGenerator`, there are more customization options. The work-around to keep this test working is to select the last "Custom" button that appears now - which happens to be the one for setting the min/max values on the Y-axis.

**Review by danxmoran — APPROVED** [review]
